content/blogs/2024-01-08-sentry-plugin.md

```diff
@@ -0,0 +1,550 @@
+
+![sentry arrows](/blogs/2024-01-08-sentry-plugin/sentry_payload_extras_arrow.png)
+
+In this example, the link URL has been composed so that it will take you right to the Kestra page for this execution, where you can view any of the relevant information:
+
+![sentry kestra](/blogs/2024-01-08-sentry-plugin/sentry_link_to_kestra.png)
+
+If your use case for Sentry is to actually capture failures, then this link can be composed like this, so that clicking it will take you straight to the logs tab of the executions details:
+
+`Link`: http://localhost:8080/ui/executions/{{flow.namespace}}/{{flow.id}}/{{execution.id}}/logs
+
+Will produce:
+
+http://localhost:8090/ui/executions/sentry_article/sentry_info_task/1TFBhMb71oc30cvfActPxp/logs
+
+And finally, when you execute the flow, click on the Gantt line item and Kestra will show you the JSON that was sent to Sentry:**
+
+![sentry produce](/blogs/2024-01-08-sentry-plugin/kestra_just_send_log_link_json.png)
+
+### SentryAlert Example
+
+This flow, [Purge execution data including logs, metrics and outputs on a schedule](https://demo.kestra.io/ui/blueprints/community/202). This flow will purge all execution data older than 1 month, including logs, metrics, and outputs. On success or failure, an issue will be sent to Sentry. It demonstrates the minimal use of the payload property:
+
+```yaml
+id: purge_storage
+namespace: sentry_article
+description: Based on https://demo.kestra.io/ui/blueprints/community/202
+
+tasks:
+  - id: clean_up_storage
+    type: io.kestra.core.tasks.storages.Purge
+    endDate: "{{ trigger.date | dateAdd(-1, 'MONTHS') }}"
+    purgeExecution: true
+    purgeLog: true
+    purgeMetric: true
+    purgeStorage: true
+    states:
+      - SUCCESS
+
+
+  - id: send_sentry_alert_on_success
+    type: io.kestra.plugin.notifications.sentry.SentryAlert
+    dsn: "{{ secret('SENTRY_DSN') }}"
+    payload: |
+      {
+          "message": "Kestra Monthly Data Purge Success",
+          "level": "info"
+      }
+
+errors:
+  - id: send_sentry_alert_on_failure
+    type: io.kestra.plugin.notifications.sentry.SentryAlert
+    dsn: "{{ secret('SENTRY_DSN') }}"
+    payload: |
+      {
+          "message": "Kestra Monthly Data Purge FAIL",
+          "level": "error"
+      }
+
+triggers:
+  - id: monthly
+    type: io.kestra.core.models.triggers.types.Schedule
+    cron: "0 9 1 * *" # every month at 9am on the 1st day of the month
+```
+
+### SentryExecution Task
+`type: io.kestra.plugin.notifications.sentry.SentryExecution`
+
+This task demonstrates a really cool feature of Kestra: triggers that trigger other triggers. Let me explain.
+
+
+If you have just one flow with some non-zero likelihood of failure - you add a SentryAlert like the examples above. But what do you do when you have 10 or 20 flows? You definitely do not want to have to maintain 20 individual SentryAlert tasks. **That’s when you use the SentryExecution task because it can triggered by any other flow.**
```

Comment:
```suggestion
If you have just one flow with some non-zero likelihood of failure - you add a `SentryAlert` like the examples above. But what do you do when you have 10 or 20 flows? You definitely do not want to have to maintain 20 individual `SentryAlert` tasks. **That’s when you use the `SentryExecution` task because any other flow can trigger it.**
```